flow/designs/asap7/megaboom/io.tcl

```diff
@@ -0,0 +1,105 @@
+proc natural_sort {list} {
+    return [lsort -command natural_compare $list]
+}
+# Custom comparison function
+proc natural_compare {str1 str2} {
+    set list1 [split $str1]
+    set list2 [split $str2]
+    set len [expr {min([llength $list1], [llength $list2])}]
+    for {set i 0} {$i < $len} {incr i} {
+        set part1 [lindex $list1 $i]
+        set part2 [lindex $list2 $i]
+        if {$part1 ne $part2} {
+            if {[string is integer -strict $part1] && [string is integer -strict $part2]} {
+                return [expr {$part1 - $part2}]
+            } else {
+                return [string compare $part1 $part2]
+            }
+        }
+    }
+    return [expr {[llength $list1] - [llength $list2]}]  ;# If all parts are equal, compare by length
+}
+
+
+proc match_pins { regex } {
+    set pins {}
+    # The regex for get_ports is not the tcl regex
+    foreach pin [get_ports -regex .*] {
+        set input [get_property $pin name]
+        # We want the Tcl regex
+        if {![regexp $regex $input]} {
+            continue
+        }
+        lappend pins [get_property $pin name]
+    }
+    return [natural_sort $pins]
+}
+
+set block [ord::get_db_block]
+
+set m4pitch 0.048
+set m5pitch 0.048
+
+array set pinGroups [list]
+set pinGroups(top) [ list \
```

Comment:
these are legal, but surprising tcl identifiers. The use if parenthesis I have not seen in tcl  identifiers.

Use something more tcl idiomatic to improve readability?

Reply:
Sure